src/cromshell/utilities/cromshellconfig.py

```diff
@@ -122,6 +127,32 @@ def resolve_cromwell_config_server_address(server_user=None, workflow_id=None):
                 )
 
 
+def __ensure_correct_submission_database_format(
+    submission_file_path: Union[str, Path]
+) -> bool:
+    """Read the first line of the submission database. If not tab-delimited (old format)
+    then update the database so it is tab-delimited."""
+
+    old_format = False
+
+    with open(submission_file_path, "r") as f:
+        first_line = f.readline()
+        if "\t" not in first_line:
+            old_format = True
+
+    if old_format:
```

Comment:
This has the problem that if I start 2 cromshells at the same time it can easily clobber itself.  Although that's sort of a general problem with how cromshell works really....

It's often best to do these operations so the update is atomicish in case it fails midway through it doesn't destroy the db.  Usually the pattern is read -> copy into new file -> move new file into old file.  I think @bshifaw knows a magic library that makes that automatic with an "update" file option that handles it all in the background.

Reply:
This is a really good point